Update and expand docs and code comments related to Plutip testing (#1504)
Plutonomicon/cardano-transaction-lib: doc/test-utils.md

Particular values can be constructed with utility functions, as demonstrated in the [ContractTestUtils example](../examples/ContractTestUtils.purs) (see `mkAssertions`).

> An example on using defined checks in the tests is an [assertions test suite](../test/Plutip/Contract/Assert.purs).

Reviewer: ```suggestion
An example for using checks in the tests is [here](../test/Plutip/Contract/Assert.purs).
```
